src/webui/www/public/scripts/dynamicTable.js

```diff
@@ -777,23 +777,39 @@ var TorrentsTable = new Class({
             this.columns['state_icon'].updateTd = function (td, row) {
                 var state = this.getRowValue(row);
 
-                if ((state === "forcedDL") || (state === "metaDL"))
-                    state = "downloading";
-                else if (state === "allocating")
-                    state = "stalledDL";
-                else if (state === "forcedUP")
-                    state = "uploading";
-                else if (state === "pausedDL")
-                    state = "paused";
-                else if (state === "pausedUP")
-                    state = "completed";
-                else if ((state === "queuedDL") || (state === "queuedUP"))
-                    state = "queued";
-                else if ((state === "checkingDL") || (state === "checkingUP") ||
-                        (state === "queuedForChecking") || (state === "checkingResumeData"))
-                    state = "checking";
-                else if ((state === "unknown") || (state === "error") || (state === "missingFiles"))
-                    state = "error";
+                switch (state) {
+                    case "forcedDL":
+                    case "metaDL":
+                        state = "downloading";
+                        break;
+                    case "allocating":
+                        state = "stalledDL";
+                        break;
+                    case "forcedUP":
+                        state = "uploading";
+                        break;
+                    case "pausedDL":
+                        state = "paused";
+                        break;
+                    case "pausedUP":
+                        state = "completed";
+                        break;
+                    case "queuedDL":
+                    case "queuedUP":
+                        state = "queued";
+                        break;
+                    case "checkingDL":
+                    case "checkingUP":
+                    case "queuedForChecking":
+                    case "checkingResumeData":
+                        state = "checking";
+                        break;
+                    case "unknown":
+                    case "error":
+                    case "missingFiles":
```

Comment:
add `default:` here? so state will always match something.

Reply:
I've added the default case

Reply:
There was a default case, but not in the merged commits.

Reply:
It must have gotten lost in the rebase. I’ll add it back later tonight.

Reply:
@Piccirello don't forget about this. I want to release v4.0.4 soon.

Reply:
@sledgehammer999 I've addressed this in #8286